apps/gro/controllers/confirm.js

```diff
@@ -1,36 +1,83 @@
 'use strict';
 
 const BaseController = require('hof').controllers.confirm;
-const path = require('path');
-const i18n = require('hof').i18n;
-const Model = require('../../common/models/email');
+const _ = require('lodash');
+const config = require('../../../config');
+const EmailService = require('../../common/services/email');
+const fields = require('../fields');
+
+delete require.cache[require.resolve('../steps')];
+const stepConfig = require('../steps');
 
 module.exports = class ConfirmController extends BaseController {
 
-  saveValues(req, res, callback) {
-    const locali18n = i18n({
-      path: path.resolve(
-        __dirname, '../translations/__lng__/__ns__.json'
+  formatStepsForEmail(data) {
+    return _(stepConfig)
+      .reject(step => !step.locals || !step.fields || _.every(
+        step.fields,
+        field => data[field] === undefined)
       )
-    });
-
-    locali18n.on('ready', () => {
-
-      const subject = locali18n.translate('pages.email.information.subject');
-      const data = {
-        values: Object.assign({}, req.sessionModel.toJSON(), {
-          reportDate: (new Date()).toISOString()
-        }),
-        email: req.sessionModel.get('email-text')
-      };
-
-      const model = new Model(data);
+      .groupBy(step => step.locals.section)
+      .map((steps, section) => ({
+          section,
+          fields: _(steps)
+            .pluck('fields')
+            .flatten()
+            .reject(field => fields[field].includeInEmail === false)
+            .map(field => ({
+              field,
+              value: data[field]
+            }))
+            .value()
```

Comment:
It is reformatting the form data so mustache can use to generate the email.  It is a straightforward procedural chain, if we split it out into functions I think it will lose a lot of readability. Instead I can add comments to show the format of the data before and the desired format after processing?

Also - this is throwaway code, at some point the steps will be arranged by section so a lot of this can be removed